hibernate-core/src/main/java/org/hibernate/type/format/JsonValueJDBCTypeAdapter.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.type.format;
+
+import org.hibernate.type.descriptor.WrapperOptions;
+import org.hibernate.type.descriptor.java.JavaType;
+import org.hibernate.type.descriptor.jdbc.JdbcType;
+
+import java.sql.SQLException;
+
+/**
+ * Adapter for JSON value on given JDBC types.
+ */
+public interface JsonValueJDBCTypeAdapter {
```

Comment:
Since this adapter is specific to the `JsonDocumentReader`, why not inline that there?

Reply:
The adapter indeed relies on the reader but is used as a mapper between values fetched by the reader and the upper layers. Readers only know about JSON types. As such, it seems natural that adapters stay beside readers.
Make sense?

Reply:
Hmm, I'll have to think a bit about this.